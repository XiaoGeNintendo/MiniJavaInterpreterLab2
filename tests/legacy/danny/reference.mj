// Checks the reference

int main() {
    int[][] a = null;
    int[] b = new int[] {1, 2, 3, -1};

    a = new int[][] {b, b};
    println(a);

    b[3] = 4;
    println(a);

    a[1] = new int[] {5, 6, 7, 8};
    println(a);

    b = null;
    println(a);

    return 0;
}

/*
[[1, 2, 3, -1], [1, 2, 3, -1]]
[[1, 2, 3, 4], [1, 2, 3, 4]]
[[1, 2, 3, 4], [5, 6, 7, 8]]
[[1, 2, 3, 4], [5, 6, 7, 8]]
Process exits with the code 0.
*/