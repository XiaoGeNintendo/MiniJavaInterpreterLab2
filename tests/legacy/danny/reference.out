[[1, 2, 3, -1], [1, 2, 3, -1]]
[[1, 2, 3, 4], [1, 2, 3, 4]]
[[1, 2, 3, 4], [5, 6, 7, 8]]
[[1, 2, 3, 4], [5, 6, 7, 8]]
Process exits with the code 0.